### ISSUE TITLE
[Request] Show more characters in button badge.

### ISSUE THREAD

**Issue by user0022** [l10nelw/winger#56]
Hello!
If possible, I would like to see more characters in button badge, as for now I can see only 3 or 4, which I find is a bit too few...
Yet one can see that there is apparently still free space on the button.

I guess what could possibly be done could be:
- Lower a bit font size, even if text is already rather small (and/or choose a narrower font type).
- Expand badge field to the maximum on the left (on the right, it seems already to the maximum).
That would put on the button rather a tab/sticker/label than a badge, well a bit less beautiful but more useful! 
- Remove margin on left between text and border on this field (on the right, there doesn't seem to be a one so nothing to remove).

With these, one could go maybe up to 7 or 8 characters.

Maybe even (?):
- Put badge text on 2 lines, to double its numbers of characters.

**Comment by user0022** [comment]
A small suggestion also for other users, to maybe help writing in the badge (at least it is what I did today and I thought maybe not it could be useful for someone else).

Personnally I can sort my windows in say 4 or 5 domains, each one having several "sub-domains" (maybe also some windows can be unique).
I "can", but I rather "prefer", because it seems to help finding them.
To be clearer, I could have say 4 windows about computing, called C, 1 for OS, 1 for browsers, 1 for repair, 1 for misc. (all the rest, or not yet sorted in right window). And 3 others for my personnal CEO business (I'm not a CEO :) it's just an example), called B, 1 for communication with my 1000 employees, 1 for R&D, 1 for bills.

So as the badge, which takes the 3 or 4 first window title characters, seems useful, I write windows names without forgetting what they will become in the badges.
So at the beginning I had done for example:
B|R&D
C|Browsers
B|Bills
(etc)
That became as badges something like:
B|R&
C|Br
B|Bi
(etc)

But I thought it could maybe be a bit improved, to be a bit clearer and to gain some space/pixels to display more title text.
So I tried to put a "special character" for the "first domain" name, as the badge field accepts them.
I took the ones present in my OS Lubutu, in (translation): Main menu > Accessories > Characters > Common (on the list on the left).
I've not tried everything, for the moment result is not wonderful, but it makes gain some pixels in general (depending on letters in fact).
🅑 or 🅱 don't work for me,in the badge they become too small and hard to read.
So I took 𝗕. Not differentiated enough, seems to be a letter of the word, but being bold like that differentiates a bit yet, and so permits me to remove the |, and then to gain some pixels.
That can with luck give as a result something like:
𝗕R&D
𝗖Bro
𝗕Bil
(etc)

Note that in "Characters" table there are also icons/smileys (well they are in black and white, maybe in next software version they will be in color).
So for a music window for instance I coud put things like 🎸🎷🎧🎵.
I've noticed also that in the badge field Ctrl+Z and Ctrl+Y work (undo/redo), a bit funny, that can help gain a bit time in text modifications.

**Comment by l10nelw** [comment]
Hi @user0022, unfortunately there is nothing we can do about a button badge's width, height or font size. They are not really meant for more than small numeric or text content. Winger just uses them as a strong "hint" for identifying a window (one of many ways to do so; see "Where do names appear?" in the **/help** page).

You can also refer to "Naming tips" in the **/help** page for some options to improve badge usefulness given its limitations. They can certainly accommodate your suggestions, I think! Thank you for taking the time to share them.

**Comment by user0022** [comment]
Thank you for clear explanations, I went to the Hepl page (which I should have done first) and found indeed that you had explained yourself the smileys/emojis possibility, or the regex.
It prevented me also to open a useless request for title preface removal option, as you already had done it.

Yet I've imagined a maybe possible workaround, who knows!
If changing addon icons in the browser toolbar would be possible (like I see for example uBlock Origin turning its color from red to yellow sometimes), plus doing it according to each window, then when giving a new name to a window, the addon could create a new icon image, with same original background, but with the name writen on it another way, so this time without the badge limitations.
But maybe totally impossible but well.
